AmplifyTools/AmplifyXcode/Sources/AmplifyXcode/Support/ArgumentParser+CLICommandEncodable.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import ArgumentParser
+
+/// The following extensions on ArgumentParser commands parameters property wrappers
+/// help us providing a hook to generate a JSON representation of a CLI command.
+/// As for now `@PropertyWrappers` APIs to access enclosing type are still "private", so the passed
+/// `registry` allows the property to reference an external type.
+/// `Argument` has been left out on purpose as we'd rather use options and flags for clarity of use.
+extension Option where Value: ExpressibleByArgument {
+    init(wrappedValue: Value, name: String, help: String, _ registry: CLICommandEncodableParametersRegistry) {
+        self.init(
+            wrappedValue: wrappedValue,
+            name: .customLong(name),
+            parsing: .next,
+            completion: nil,
+            help: ArgumentHelp(help)
+          )
+        let type = String(describing: Value.self)
+        registry.register(param: .option(name: name, type: type, help: help))
```

Comment:
As above, I wonder if the registry is necessary or whether we can just make `params` a `Set`